examples/step-6/step-6.cc

```diff
@@ -284,10 +284,7 @@ void Step6<dim>::assemble_system()
 
   std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
 
-  typename DoFHandler<dim>::active_cell_iterator cell =
-                                                   dof_handler.begin_active(),
-                                                 endc = dof_handler.end();
-  for (; cell != endc; ++cell)
+  for (const auto cell : dof_handler.active_cell_iterators())
```

Comment:
```suggestion
  for (const auto &cell : dof_handler.active_cell_iterators())
```

Reply:
IIRC clang does actually complain about this.

Reply:
I have changed this to `const auto &cell`. But I disagree that the former version was wrong. (Whether you copy or keep a reference to an iterator/accessor doesn't make that much of a difference.)

Reply:
In the same spirit:
 - we advertise `for (auto cell` in `base/iterator_range.h`
 - we also have `for (auto cell` in `step-18`, `step-2`, `step-31`, `step-49`, `step-60`, `step-62`.
We should unify this! :-)

Reply:
See also: https://github.com/dealii/dealii/pull/6781

Reply:
What you wrote is not wrong (nor is this particularly important).

My impression from #6781 was that `const auto &cell` was the preferred version because of some compiler warning at some point.